src/engine/QMLEngine.js

```diff
@@ -493,12 +498,23 @@ class QMLEngine {
 
   // Return a path to load the file
   $resolvePath(file) {
+    if (this.$rootBasePath === undefined || this.$basePath === undefined) {
+      return file;
+    }
+
+    // "Root" references in QML should still be relative to the root QML file
+    if (file.indexOf("/") === 0 && this.$rootBasePath !== undefined) {
```

Comment:
> I am not sure if we should store $rootBasePath though, as we could just do...

I don't think that will work.  `$rootBasePath` is only set when then initial QML file is loaded from the `data-qml` attribute of the `body` tag, whereas `$basePath` is set to the path of whatever QML component file is currently being evaluated (and, in this change, in a few more places to ensure that the correct `$basePath` is used for evaluating relative paths).

So, for example, if you have a directory structure like:

```
http://www.example.com
/index.html
/test/test.qml
/test/foo/Bar.qml
/test/foo/image.png
```

And `index.html` has:

```
<body data-qml="test/test.qml">
```

test.qml:

```
import "foo"

Bar {
}
```

Bar.qml

```
import QtQuick 2.0
Item {
    property url path: "image.png" // $basePath would be set to "test/foo" when this evaluated so that the path would resolve to "test/foo/image.png".  $rootBasepath would always be "test/".
}
```

> Could you provide an example where this would break?

In the above example, if the path in `Bar.qml` was `/image.png`, it will now resolve to `http://www.example.com/test/image.png`. Whereas `/image.png` in the normal HTML/CSS-case would resolve to `http://www.example.com/image.png`.

In the realm of file-based QML, `/image.png` would resolve to `file:///image.png`, which to me feels more like `http://www.example.com/image.png`. So I think it makes the most sense to leave out the `$rootBasePath` stuff.  I think what I was trying to accomplish is more of a job for the `qrc` system.

I feel the best approach going forward would be to expand upon the `qrc` system to behave more like QML (e.g. if a Component file is within the `qrc` file system, all paths are relative to the `qrc` file system unless fully resolved). And add some support for "storing" images in the `qrc` filesystem (probably just as an HTML-relative URL, maybe with an option for base64 encoding if that's something someone wants).